pkg/image/containerd/daemon_provider.go

```diff
@@ -446,6 +450,16 @@ func (p *daemonImageProvider) saveImage(ctx context.Context, client *containerd.
 	return tempTarFile.Name(), nil
 }
 
+func (p *daemonImageProvider) deleteImage(ctx context.Context, client *containerd.Client, resolvedImage string) error {
+	// Remove the image
+	if err := client.ImageService().Delete(ctx, resolvedImage, nil); err != nil {
+		return fmt.Errorf("failed to delete image %s: %w", resolvedImage, err)
+	}
+
+	log.Infof("deleted image %s from containerd", resolvedImage)
```

Comment:
This provider is used for docker, podman, etc., so should use the `p.Name()` function instead of hardcoded `containerd`